src/components/socketComponent.js

```diff
@@ -1,41 +1,52 @@
-import React, { Component, Fragment } from 'react'
+import React, { Component, Fragment } from 'react';
 import PropTypes from 'prop-types';
 
 import openSocket from 'socket.io-client';
 
 class SocketComponent extends Component {
-	constructor(props){
-		super(props)
-		const socket = openSocket(props.socketAddress)
-		const listeners = {}
-		
-		if(props.onConnect){
-			const [ ev, val ] = props.onConnect
-			listeners[ev] = socket.on('connect', () => socket.emit(ev, val))
-		}
+  state = {
+    listeners: [],
+    socket: null,
+    socketAddress: null
+  };
+  componentDidMount() {
+    const { onConnect, socketAddress } = this.props;
+    const socket = openSocket(socketAddress);
+    const listeners = {};
 
-		props.listeners.forEach(listener => {
-			listener.name.forEach(name => {
-				listeners[name] = socket.on(name, val => listener.onEvent(val, socket)) // socket sent in as second argument so that you can make emit another event
-			})
-		})
-		this.state = {socket, ...listeners}
-	}
+    if (onConnect) {
+      const [event, val] = onConnect;
+      listeners[event] = socket.on('connect', () => socket.emit(event, val));
+    }
+    this.props.listeners.forEach(listener => {
```

Comment:
This nested `forEach` seems particularly confusing to me. I'm more comfortable with nested `for` loops, visually. What say ye?